NDTensors/src/fill.jl

```diff
@@ -0,0 +1,64 @@
+# This is a file to write generic fills for NDTensors.
+#  This includes random fills, zeros, ...
+
+function generic_randn(
+  StoreT::Type{<:Dense{ElT,DataT}}, dim::Integer=0
+) where {DataT<:AbstractArray{ElT}} where {ElT}
+  data = generic_randn(DataT, dim)
+  return StoreT(data)
+end
+
+function generic_randn(StoreT::Type{<:Dense}, dim::Integer=0)
+  return generic_randn(StoreT{default_eltype(),default_datatype(default_eltype())}, dim)
+end
+
+function generic_randn(DataT::Type{<:AbstractArray{ElT}}, dim::Integer=0) where {ElT}
+  data = DataT(undef, dim)
+  for i in 1:dim
+    @inbounds data[i] = randn(ElT)
+  end
+  return data
+end
+
+function generic_randn(DataT::Type{<:AbstractArray}, dim::Integer=0)
+  return generic_randn(DataT{default_eltype()}, dim)
+end
+
+##TODO replace randn in ITensors with generic_randn
+## and replace zeros with generic_zeros
+function randn(
+  StoreT::Type{<:Dense{ElT,VecT}}, dim::Integer
+) where {VecT<:AbstractArray{ElT}} where {ElT<:Number}
+  return StoreT(randn(ElT, dim))
+end
+
+function generic_zeros(
+  StoreT::Type{<:Dense{ElT,DataT}}, dim::Integer=0
+) where {DataT<:AbstractArray{ElT}} where {ElT}
+  data = generic_zeros(DataT, dim)
```

Comment:
I would use `datatype(storagetype)` instead of `DataT` (also change `StoreT` to `storagetype`, I've preferred that style but I know that's inconsistent throughout NDTensors right now).

Reply:
I see this case explicitly uses `DataT` to indicate to dispatch that the data type is specified so it probably makes more sense to just use ` DataT` and not `datatype(StoreT)`/`datatype(storagetype)` in this case.